src/jaqt.mjs

```diff
@@ -1,13 +1,12 @@
 /**
- * checks if data is a plain object and not null, String, Number, Boolean or Array or other classes
- * 
+ * checks if data is a wrapper object for one of the primitive types
+ *
  * @param  {mixed}  data The data to check
- * @return {Boolean}     True if data is a plain object
+ * @return {Boolean}     True if data is a primitive wrapper object or undefined
  */
-function isPlainObject(data)
+function isPrimitiveWrapper(data)
 {
-    return data?.constructor === Object 
-        || data?.constructor === undefined // object with null prototype: Object.create(null)
+    return [String, Boolean, Number, undefined].includes(data?.constructor)
```

Comment:
I think 'undefined' should not be in this list, as that means that it is in fact a 'plain' object. The reason to filter out String, Boolean and Number is to prevent trying to select/filter properties of those objects. Instead these should be handled like primitive values. This is not the case for 'undefined'.

Reply:
Got it, I just kept that from the original function thinking that `null` would be a primitive which should be included here.
Is it fine if I force-push onto my branch?

Reply:
Sure, no problem.